packages/ui/src/common/components/EditableInputList/EditableInputList.tsx

```diff
@@ -0,0 +1,124 @@
+    const values = [...questionFields]
+    values.splice(index, 1)
+    addQuestionField(values)
+  }
+
+  const handleInputChange = (index: number, event: ChangeEvent<HTMLInputElement>) => {
+    const values = [...questionFields]
+    values[index].questionField = event.target.value
+    addQuestionField(values)
+  }
+
+  const toggleLength = (index: number, shortValue: boolean) => {
+    const values = [...questionFields]
+    values[index].shortValue = shortValue
+    addQuestionField(values)
+  }
+
+  return (
+    <>
+      <HeaderWrapper>
+        <TextHuge>{title}</TextHuge>
+        <TransactionButton style="primary" size="medium" onClick={() => addFields()}>
+          <PlusIcon />
+          {buttonText}
+        </TransactionButton>
+      </HeaderWrapper>
+      {questionFields.map((questionFields, index) => (
+        <>
+          <QuestionFieldWrapper>
+            <StyledInputComponent
+              id="field-title"
+              inputSize="m"
+              required
+              message={questionFields.questionField.length > 50 ? 'Max length is 55 characters' : 'MAX 55'}
+              validation={questionFields.questionField.length > 50 ? 'invalid' : undefined}
+            >
+              <InputText
+                id="field-question"
+                value={questionFields.questionField}
+                onChange={(e) => handleInputChange(index, e)}
+                placeholder="Type question"
+              />
+            </StyledInputComponent>
+            <DeleteButtonWrapper onClick={() => removeFields(index)} size="small" square>
+              <DeleteIcon />
+            </DeleteButtonWrapper>
+          </QuestionFieldWrapper>
+          <ToggleWrapper>
+            <ToggleCheckbox
+              trueLabel={questionFields.shortValue ? <StyledLabel>Short answer</StyledLabel> : 'Short answer'}
+              falseLabel={!questionFields.shortValue ? <StyledLabel>Long answer</StyledLabel> : 'Long answer'}
+              onChange={() => toggleLength(index, !questionFields.shortValue)}
+              checked={questionFields.shortValue ?? false}
+              switchToggle
+            />
+          </ToggleWrapper>
+        </>
+      ))}
+    </>
+  )
+})
+
+export default EditableInputList
+
+const HeaderWrapper = styled.div`
+  display: flex;
+  justify-content: space-between;
+`
+
+const DeleteButtonWrapper = styled(ButtonBareGhost)`
+  padding-top: 20px;
+`
+
+const DeleteIcon = styled(BinIcon)`
+  height: 16px;
+  width: 16px;
+`
+
+const StyledInputComponent = styled(InputComponent)<{ validation?: string | undefined }>`
+  ${InputNotification} {
+    justify-content: ${({ validation }) => validation === undefined && 'end'};
+  }
+`
+
+const QuestionFieldWrapper = styled.div`
+  display: flex;
+  justify-content: space-between;
+`
+
+const ToggleWrapper = styled.div`
+  display: flex;
+  justify-content: end;
+  padding-right: 35px;
+`
+
+const StyledLabel = styled(TextMedium)`
+  font-weight: 900;
+`
```

Comment:
Please add some vertical padding (or whatever works) between the each question, and between the questions and the add button